internal/controller/drplacementcontrol_controller.go

```diff
@@ -854,15 +850,76 @@ func (r *DRPlacementControlReconciler) updateAndSetOwner(
 	usrPlacement client.Object,
 	log logr.Logger,
 ) (bool, error) {
-	if err := r.annotateObject(ctx, drpc, usrPlacement, log); err != nil {
+	var (
+		err                       error
+		updated                   bool
+		placementModified         bool
+		placementAnnotateModified bool
+		placementMetadataModified bool
+	)
+
+	// Annotate the placement object
+	if placementAnnotateModified, err = r.annotateObject(drpc, usrPlacement, log); err != nil {
 		return false, err
 	}
 
-	if err := r.updateObjectMetadata(ctx, drpc, usrPlacement, log); err != nil {
+	// Update object metadata
+	if _, placementMetadataModified, err = r.updateObjectMetadata(drpc, usrPlacement); err != nil {
 		return false, err
 	}
 
-	return r.setDRPCOwner(ctx, drpc, usrPlacement, log)
+	// Merge both modifications
+	placementModified = placementAnnotateModified || placementMetadataModified
```

Comment:
placementModified can be computed inline, I mean something like this:
	if placementAnnotateModified || placementMetadataModified {
		if err := r.updatePlacement(ctx, usrPlacement, log); err != nil {
			return false, err
		}
	}